### PR TITLE
[Resource][Js] Fix broken auto complete on edit page

### PR DESCRIPTION
**Pull request by Arminek**
| Q               | A
| --------------- | ---
| Bug fix?        | yes |
| New feature?    | no |
| BC breaks?      | no |
| License         | MIT |

### CODE REVIEW

File: src/Sylius/Bundle/CoreBundle/Form/Type/Taxon/ProductTaxonAutocompleteChoiceType.php [hunk 1]
```diff
@@ -65,8 +84,13 @@ public function configureOptions(OptionsResolver $resolver)
         $resolver->setDefaults([
             'resource' => 'sylius.taxon',
             'choice_name' => 'name',
-            'choice_value' => 'code',
+            'choice_value' => 'code'
```
Comment (pamil) [inline src/Sylius/Bundle/CoreBundle/Form/Type/Taxon/ProductTaxonAutocompleteChoiceType.php]: Why?

File: src/Sylius/Bundle/CoreBundle/Form/Type/Taxon/ProductTaxonAutocompleteChoiceType.php [hunk 2]
```diff
@@ -45,14 +56,22 @@ public function buildForm(FormBuilderInterface $builder, array $options)
...
         }
 
         if (!$options['multiple']) {
             $builder->addModelTransformer(
-                new ProductTaxonToTaxonTransformer($this->productTaxonFactory)
+                new ProductTaxonToTaxonTransformer(
+                    $this->productTaxonFactory,
+                    $this->productTaxonRepository,
+                    $options['product']
+                )
```
Comment (pamil) [inline src/Sylius/Bundle/CoreBundle/Form/Type/Taxon/ProductTaxonAutocompleteChoiceType.php]: That escalated quickly, what about?

```php
$transformer = new ProductTaxonToTaxonTransformer(...);

if ($options['multiple']) {
    $transformer = new RecursiveTransformer($transformer);
}

$builder->addModelTransformer($transformer);
```

File: src/Sylius/Bundle/CoreBundle/Tests/Form/Type/Taxon/ProductTaxonAutocompleteChoiceTypeTest.php [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+    public function it_creates_new_product_taxons_based_on_given_product_and_passed_taxon_codes()
+    {
+        $taxon = $this->prophesize(TaxonInterface::class);
+        $product = $this->prophesize(ProductInterface::class);
+
+        $taxonRepository = $this->prophesize(TaxonRepositoryInterface::class);
+
+        $this->resourceRepositoryRegistry->get('sylius.taxon')->willReturn($taxonRepository);
+        $taxonRepository->findOneBy(['code' => 'mug'])->willReturn($taxon);
+        $taxonRepository->findOneBy(['code' => 'book'])->willReturn($taxon);
```
Comment (pamil) [inline src/Sylius/Bundle/CoreBundle/Tests/Form/Type/Taxon/ProductTaxonAutocompleteChoiceTypeTest.php]: It looks wrong, the same taxon returned for both codes?

File: src/Sylius/Bundle/CoreBundle/Tests/Form/Type/Taxon/ProductTaxonAutocompleteChoiceTypeTest.php [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+        $this->productTaxonFactory->createNew()->willReturn($productTaxon);
+
+        $form = $this->factory->create(ProductTaxonAutocompleteChoiceType::class, new ArrayCollection(), [
+            'label' => 'sylius.form.product.taxons',
+            'product' => $product->reveal(),
+            'multiple' => true,
+        ]);
+
+        $form->submit('mug,book');
+        $this->assertEquals(new ArrayCollection([$productTaxon->reveal(), $productTaxon->reveal()]), $form->getData());
```
Comment (pamil) [inline src/Sylius/Bundle/CoreBundle/Tests/Form/Type/Taxon/ProductTaxonAutocompleteChoiceTypeTest.php]: And the same product taxons returned?

File: src/Sylius/Bundle/CoreBundle/Tests/Form/Type/Taxon/ProductTaxonAutocompleteChoiceTypeTest.php [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+    public function it_returns_existing_product_taxons_based_on_given_product_and_passed_taxon_codes()
+    {
+        $taxon = $this->prophesize(TaxonInterface::class);
+        $product = $this->prophesize(ProductInterface::class);
+        $productTaxon = $this->prophesize(ProductTaxonInterface::class);
+        $taxonRepository = $this->prophesize(TaxonRepositoryInterface::class);
+
+        $this->resourceRepositoryRegistry->get('sylius.taxon')->willReturn($taxonRepository);
+        $taxonRepository->findOneBy(['code' => 'mug'])->willReturn($taxon);
+        $taxonRepository->findOneBy(['code' => 'book'])->willReturn($taxon);
```
Comment (pamil) [inline src/Sylius/Bundle/CoreBundle/Tests/Form/Type/Taxon/ProductTaxonAutocompleteChoiceTypeTest.php]: Same as above.

File: src/Sylius/Bundle/CoreBundle/Tests/Form/Type/Taxon/ProductTaxonAutocompleteChoiceTypeTest.php [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+        $this->productTaxonRepository->findOneBy(['product' => $product, 'taxon' => $taxon])->willReturn($productTaxon);
+
+        $form = $this->factory->create(ProductTaxonAutocompleteChoiceType::class, new ArrayCollection(), [
+            'label' => 'sylius.form.product.taxons',
+            'product' => $product->reveal(),
+            'multiple' => true,
+        ]);
+
+        $form->submit('mug,book');
+        $this->assertEquals(new ArrayCollection([$productTaxon->reveal(), $productTaxon->reveal()]), $form->getData());
```
Comment (pamil) [inline src/Sylius/Bundle/CoreBundle/Tests/Form/Type/Taxon/ProductTaxonAutocompleteChoiceTypeTest.php]: Same as above.

**Comment by pjedrzejewski** [comment]
Thanks Arek!

**Pull request merged**